pkg/otlp/metrics/metrics_translator.go

```diff
@@ -202,9 +202,17 @@ func (t *Translator) mapNumberMonotonicMetrics(
 			continue
 		}
 
-		if dx, ok := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val); ok {
+		dx, firstPoint, dropPoint := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val)
+		if dropPoint {
+			t.logger.Debug("Dropping point: timestamp is older or equal to timestamp of previous point received", zap.String(metricName, pointDims.name))
+			continue
+		}
+
+		if !firstPoint {
 			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, dx)
 		} else if i == 0 && t.shouldConsumeInitialValue(startTs, ts) {
+			// We only compute the first point in the timeseries if it is the first value in the datapoint slice.
+			// Todo: Investigate why we don't compute first val if i > 0 and add reason as comment.
```

Comment:
@songy23 We may have discussed this when you aded this functionality, but I don't remember. Do you recall the rationale behind this?

Reply:
We only consume the initial value for cumulative metrics. We do so by calculating the delta as if the previous value is 0. When `i>0` we just calculate the real delta (`pts[i] - pts[i-1]`)

Reply:
But if there is a reset within a datapoint slice, e.g. values of `[]int64{10, 30, 5, 20}`.

In this case, there is a reset at index 2 and we can't compute the diff because it would be negative. Because of the `i>0`, we don't compute 5 as the initial value, but only put the point in the cache. Is this the intended behaviour ? Or should we consume 5 as the initial value here ?

Reply:
These two tests: https://github.com/DataDog/opentelemetry-mapping-go/blob/de6046116470f8e2367b9a38213a7bafe53eee0f/pkg/otlp/metrics/metrics_translator_test.go#L350-L417 show how it currently is.

**TestMapIntMonotonicWithRebootWithinSlice**
- datapoint slice 1: `[]int64{0, 30, 0, 20}`

**TestMapIntMonotonicWithRebootBeginningOfSlice**
- datapoint slice 1: `[]int64{10}`
- datapoint slice 2: `[]int64{5, 30}`

You can see that in the first test, the value at reset is not consumed as first value because it is within the slice, whereas in the second test the value at reset is consumed as first value because it is at index 0 of the slice.

Reply:
> But if there is a reset within a datapoint slice, e.g. values of []int64{10, 30, 5, 20}.

That's a different case. It is not handled in this `else if` block (it may be caught by the cache calculation but I'm not 100% sure). This block `else if i == 0 && t.shouldConsumeInitialValue(startTs, ts)` only takes care of the first value in the slice.

Reply:
> it may be caught by the cache calculation

It is caught by the cache calculation (`dx < 0`), but the point does not get computed as an initial value, it only get's cached. This is not due to the changes in this PR, as there was the same behaviour before.

I'll create a task to investigate whether this is a feasible situation (are there ever resets within a slice), and then to update the code based on that. If it's a feasible scenario, then we should compute as first val, if it's not a feasable scenario then we should make a note in the code that this should never happen.

I don't think this should block this PR as there was the same behaviour before.